oci/oci.go

```diff
@@ -688,8 +688,11 @@ func (r *Runtime) StopContainer(ctx context.Context, c *Container, timeout int64
 	}
 
 	if timeout > 0 {
-		if err := utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, r.Path(c), "kill", c.id, c.GetStopSignal()); err != nil {
-			return fmt.Errorf("failed to stop container %s, %v", c.id, err)
+		if err = utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, r.Path(c), "kill", c.id, c.GetStopSignal()); err != nil {
+			// ref: https://github.com/kubernetes-incubator/cri-o/issues/1477
+			// just log that an error happened, and leave any further logic to
+			// waitContainerStop to avoid races
+			logrus.Debugf("failed to stop container %s, %v", c.id, err)
```

Comment:
Another option is checking for the error string: `os: process already finished`
cc: @sboeuf

Reply:
that stinks a bit to me and we don't control that string (and it can change I guess) :confused: the reason for just logging is that, however, we're just interested in stopping the container and always KILL at the end if we can't gracefully stop it so little reason in bothering to check all this

Reply:
I would fix #1477 by checking if the container process is still around from the error handling. This could help you to make the diff between a real error and a case where the container properly terminated.
I think we still want to get an error in the case we have a real error.

Reply:
Isn't that still subject to the same race?

Reply:
Nope, because you are in the error handling case here. This means your `kill` already failed for some reason. Now you want to find out why, and checking if the process is still around should tell you what was the reason.

Reply:
> Another option is checking for the error string: `os: process already finished`

I agree that this is an unfortunate cludge, but we already do it (since #1445).  Perhaps we can port some more of this into `pkg/findprocess` to at least stay DRY with that comparison?  I'd rather look at the string than log instead of error for *other* errors.